apps/minifront/src/state/send/index.ts

```diff
@@ -122,10 +132,28 @@ export const createSendSlice = (): SliceCreator<SendSlice> => (set, get) => {
         });
       }
     },
+    setGasPrices: prices => {
+      set(state => {
+        state.send.gasPrices = prices;
+      });
+    },
+    setStakingToken: stakingToken => {
+      set(state => {
+        state.send.stakingToken = stakingToken;
+      });
+    },
   };
 };
 
-const assembleRequest = ({ amount, feeTier, recipient, selection, memo }: SendSlice) => {
+const assembleRequest = ({
+  amount,
+  feeTier,
+  recipient,
+  selection,
+  memo,
+  gasPrices,
+  stakingToken,
+}: SendSlice) => {
   const spendOrOutput:
     | PartialMessage<TransactionPlannerRequest_Spend>
     | PartialMessage<TransactionPlannerRequest_Output> = {
```

Comment:
suggestion: These lines seem to be the source of some funky typing downstream. Would recommend simplifying this to:

```ts
interface SpendOrOutput { 
  address: Address; 
  value: Value 
}

  const spendOrOutput: SpendOrOutput = {
    address: new Address({ altBech32m: recipient }),
    value: new Value({
      amount: toBaseUnit(
        BigNumber(amount),
        getDisplayDenomExponentFromValueView.optional(selection?.balanceView),
      ),
      assetId: getAssetIdFromValueView(selection?.balanceView),
    }),
  };
```

and propagating to all things downstream

Reply:
modified type signature